MissionTable.cpp

```diff
@@ -28,30 +28,72 @@ void WriteBoolCell(bool boolean, std::streamsize cellWidthInChars);
 
 std::string_view ConvertMissionTypeToString(MissionTypes missionType);
 
-constexpr std::array<std::string_view, 7> columnTitles{
-	"DLL NAME",
-	"TYP",
-	"#",
-	"U",
-	"MAP NAME",
-	"TECH TREE NAME",
-	"MISSION DESCRIPTION",
+struct LegendEntry
+{
+	std::string_view key;
+	std::string_view description;
+};
+
+constexpr std::array<LegendEntry, 7> columnTitles{
+	LegendEntry {"DLL NAME", "Name of mission's dll file"},
+	LegendEntry {"TYP", "Mission type"},
+	LegendEntry {"#", "Maximum supported count of combined human and AI players"},
+	LegendEntry {"U", "Is mission unit only"},
+	LegendEntry {"MAP NAME", "Map filename used by the mission"},
+	LegendEntry {"TECH TREE NAME", "Name of the tech tree used by the mission"},
+	LegendEntry {"MISSION DESCRIPTION", "Description of the mission displayed in Outpost 2"}
 };
+
 constexpr std::array<std::streamsize, 7> columnWidths{ 9, 4, 2, 2, 18, 24, 1 };
 
-constexpr std::array<std::string_view, 8> missionTypes{
+constexpr std::array<LegendEntry, 8> missionTypes{
 	// Single player
-	"Col",
-	"Dem",
-	"Tut",
+	LegendEntry {"Col", "Colony"},
+	LegendEntry {"Dem", "Demo"},
+	LegendEntry {"Tut", "Tutorial"},
 	// Multiplayer
-	"MLR",
-	"MSP",
-	"MRR",
-	"MM",
-	"ML",
+	LegendEntry {"MLR", "Multiplayer, Land Rush"},
+	LegendEntry {"MSP", "Multiplayer, Space Race"},
+	LegendEntry {"MRR", "Multiplayer, Resource Race"},
+	LegendEntry {"MM", "Multiplayer, Midas"},
+	LegendEntry {"ML", "Multiplayer, Last One Standing"}
 };
 
+template<std::size_t size>
+std::size_t FindMaxKeyWidth(const std::array<LegendEntry, size>& container) 
+{
+	return std::max_element(container.begin(), container.end(),
+		[](const LegendEntry& a, const LegendEntry& b)
+		{
+			return a.key.size() < b.key.size();
+		})->key.size();
+}
+
+template<std::size_t size>
+void WriteLegendPortion(std::string_view title, const std::array<LegendEntry, size>& container)
+{
+	std::cout << title << std::endl;
+
+	auto cellWidth = FindMaxKeyWidth(container);
+	for (const auto& entry : container) {
+		std::cout << std::setw(cellWidth) << std::left << entry.key << " = " << entry.key << std::endl;
+	}
+
+	std::cout << std::endl;
+}
+
+void WriteLegend()
+{
+	WriteLegendPortion("MISSION TABLE LEGEND", columnTitles);
+
+	// Add campaign to the top of mission types
+	std::array<LegendEntry, missionTypes.size() + 1> missionTypesWithCampaign{ LegendEntry{"Cam", "Campaign"} };
```

Comment:
I couldn't figure out a better way to include campaign in the legend. Feels like a hack.

Reply:
Actually, maybe we can just add that entry to the original array, and then adjust the indexing to account for it. It might even simplify the indexing.

Reply:
Issue opened for future work